pkg/vsphere/performance/host_metrics.go

```diff
@@ -64,17 +63,41 @@ type HostMetricsInfo struct {
 	CPU    HostCPU
 }
 
-// HostMetrics returns CPU and memory metrics for all ESXi hosts in the input session's cluster.
-func HostMetrics(op trace.Operation, session *session.Session) (map[*object.HostSystem]*HostMetricsInfo, error) {
-	if session == nil {
+// HostMetrics returns CPU and memory metrics for all ESXi hosts in a cluster
+// via implementation of the MetricsProvider interface.
+type HostMetricsProvider struct {
+	Session *session.Session
```

Comment:
Should we export `Session` if the caller can supply it in  `NewHostMetricsProvider()`?

Reply:
No, we shouldn't. Good catch, that was left over from a previous iteration where I was accessing the session from outside of the package before I had the design figured out. Fixed!